lib/services/ngsi/subscriptionService.js

```diff
@@ -126,11 +126,12 @@ function subscribe(device, triggers, content, callback) {
         store = false;
     }
 
-    if (device.cbHost) {
+    if (device.cbHost && device.cbHost.indexOf('://') !== -1) {
+        options.uri = device.cbHost + '/v1/subscribeContext';
+    } else if (device.cbHost && device.cbHost.indexOf('://') === -1) {
         options.uri = 'http://' + device.cbHost + '/v1/subscribeContext';
     } else {
-        options.uri = 'http://' + config.getConfig().contextBroker.host + ':' +
-            config.getConfig().contextBroker.port + '/v1/subscribeContext';
+        options.uri = config.getConfig().contextBroker.url + '/v1/subscribeContext';
```

Comment:
contextBroker.url was the one introduced in PR https://github.com/telefonicaid/iotagent-node-lib/pull/584, isn't it?

Reply:
yes